SLMPhi3ChunkingSkill/code/slm-chunking-flow/save_to_ai_search_vectordb.py

```diff
@@ -0,0 +1,146 @@
+
+from azure.core.credentials import AzureKeyCredential
+from azure.search.documents import SearchClient
+from azure.search.documents.indexes import SearchIndexClient
+from azure.search.documents.indexes.models import (
+    SearchField,
+    SearchIndex,
+    SearchFieldDataType,
+    VectorSearch,
+    HnswAlgorithmConfiguration,
+    VectorSearchAlgorithmMetric,
+    ExhaustiveKnnAlgorithmConfiguration,
+    ExhaustiveKnnParameters,
+    VectorSearchProfile,
+    AzureOpenAIVectorizer,
+    AzureOpenAIParameters,
+    SemanticConfiguration,
+    SemanticSearch,
+    SemanticPrioritizedFields,
+    SemanticField
+)
+
+from promptflow.core import tool
+from promptflow.connections import AzureOpenAIConnection,CognitiveSearchConnection
+
+
+def initSearch(service_endpoint : str ,index_name: str,credential: AzureKeyCredential,aoai_conn: AzureOpenAIConnection):
+  # Create a search index
+  index_client = SearchIndexClient(
+      endpoint=service_endpoint, credential=credential)
+  
+
+
+  fields = [
+      SearchField(name="id",type=SearchFieldDataType.String, key=True, sortable=True, filterable=True, facetable=True),
+      SearchField(name="chunking", type=SearchFieldDataType.String),
+      SearchField(name="chunking_vector", type=SearchFieldDataType.Collection(SearchFieldDataType.Single),searchable=True, vector_search_dimensions=1536, vector_search_profile_name="myHnswProfile")
+  ]
+
+  vector_search = VectorSearch(
+    algorithms=[
+        HnswAlgorithmConfiguration(
+            name="myHnsw"
+        )
+    ],
+    profiles=[
+        VectorSearchProfile(
+            name="myHnswProfile",
+            algorithm_configuration_name="myHnsw",
+            vectorizer="myVectorizer"
+        )
+    ],
+    vectorizers=[
+        AzureOpenAIVectorizer(
+            name="myVectorizer",
+            azure_open_ai_parameters=AzureOpenAIParameters(
+                resource_uri=aoai_conn.api_base,
+                deployment_id="EmbeddingModel",  
+                api_key=aoai_conn.api_key,  
+                model_name="text-embedding-ada-002"
+            )
+        )
+    ]
+  )
+
+#   vector_search = VectorSearch(
```

Comment:
If it's necessary to keep those commented-out lines, please include a comment saying why. Otherwise, please remove.